clusterctl/validation/validate_cluster_api_objects_test.go

```diff
@@ -0,0 +1,242 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package validation
+
+import (
+	"testing"
+
+	"github.com/kubernetes-incubator/apiserver-builder/pkg/test"
+	"k8s.io/api/core/v1"
+	meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/cluster-api/pkg/apis"
+	"sigs.k8s.io/cluster-api/pkg/apis/cluster/common"
+	"sigs.k8s.io/cluster-api/pkg/apis/cluster/v1alpha1"
+	"sigs.k8s.io/cluster-api/pkg/apis/cluster/v1alpha1/testutil"
+	"sigs.k8s.io/cluster-api/pkg/client/clientset_generated/clientset"
+	"sigs.k8s.io/cluster-api/pkg/openapi"
+)
+
+func TestGetClusterObject(t *testing.T) {
+	testClusterName := "test-cluster"
```

Comment:
This function has three groupings of test cases, No cluster, one cluster, and two clusters. Tests should be for a single case or a single table driven set of cases all in the same general area testing the same method. It will be extremely difficult for a future developer to understand and make changes to these tests.

Please refactor this into three different test methods.